packages/schematics/src/command-line/workspace-results.ts

```diff
@@ -0,0 +1,55 @@
+import * as fs from 'fs';
+import { readJsonFile, writeJsonFile } from '../utils/fileutils';
+import { unlinkSync } from 'fs';
+
+const RESULTS_FILE = '.nx-results';
+
+interface NxResults {
+  command: string;
+  results?: { [key: string]: boolean };
+}
+
+export class WorkspaceResults {
+  private startedWithFailedProjects: boolean;
+  private commandResults: NxResults;
+
+  constructor(private command: string) {
+    const resultsExists = fs.existsSync(RESULTS_FILE);
+    this.commandResults = resultsExists
+      ? readJsonFile(RESULTS_FILE)
+      : {
+          command,
+          results: {}
+        };
+    this.startedWithFailedProjects = this.commandResults.command === command;
```

Comment:
I would make it explicit that if the command isn't the same as what is stored, we should ignore what is stored in the file. I think right now it is a bit broken.

Say you have app1 and app2, and app1 has broken tests. The following shows the issue.

```
Run `yarn affected:test -- --files=package.json` (runs tests for app1 and app2)
Run `yarn affected:test -- --files=package.json --only-failed` (runs tests for app1)
Run `yarn affected:build -- --files=package.json --only-failed` (builds app1, but should build both apps because we changed the command)
```